src/object.d

```diff
@@ -3309,6 +3522,28 @@ if (isCreateOperation!(C, V) && isUpdateOperation!(U, V))
         *p = update(*p);
 }
 
+///
+@system unittest
+{
+    auto aa = ["k1": 1];
+
+    aa.update("k1", {
+        assert(0, "k1 already exists.");
+        return 0;
+    }, (ref int v) {
+        return v + 1; // update
```

Comment:
NB: we can't use `v++` here as `update` calls the function with the dereferenced `value`.
However, it also requires that the function accepts the parameter by reference. Can we please agree to fix this?